aiozk/test/test_session.py

```diff
@@ -0,0 +1,34 @@
+import asyncio
+from unittest import mock
+
+import aiozk
+
+import pytest
+
+
+def coroutine_mock(return_value=None):
+    stub = mock.Mock(return_value=return_value)
+    return asyncio.coroutine(stub)
+
+
+@pytest.fixture
+def session():
+    fake_retry_policy = mock.MagicMock()
+    fake_loop = mock.MagicMock()
+    return aiozk.session.Session('zookeeper.test', 10, fake_retry_policy, True, 30, loop=fake_loop)
+
+
+@pytest.mark.asyncio
+async def test_start_session_twice(session):
+    with mock.patch.object(session, 'ensure_safe_state', coroutine_mock()) as ensure_safe_state:
+        await session.start()
+        await session.start()
```

Comment:
It would be great to have this tests for real connection too. See comment above about reaching safe state in start.

Reply:
I can't imagine how we may test it with real connection.
What would we test?

Reply:
I'd like to have a test for the case with a preliminary return in `session.start`.

Something like call `session.start` simultaneously two times and wait for first (this will be preliminary returned session) and see what we can check here.

Reply:
that was the question: what should we check?
or do you suggest to wrap real connection with mock to check functions call order?